src/jvmTest/kotlin/test.kt

```diff
@@ -0,0 +1,96 @@
+import org.junit.jupiter.api.Test
+import org.kosat.readCnfRequests
+import java.io.File
+import kotlin.system.measureTimeMillis
+import com.github.lipen.satlib.solver.MiniSatSolver
+import org.kosat.solveCnf
+import java.nio.file.Files
+import java.nio.file.Paths
+import kotlin.math.abs
+import kotlin.math.sign
+import kotlin.streams.toList
+
+internal class DiamondTests {
+    private val projectDirAbsolutePath = Paths.get("").toAbsolutePath().toString()
+    private val format = ".cnf"
+    private val headerNames = listOf("Name:", "KoSAT time:", "MiniSAT time:", "Result:", "Solvable:")
+
+    private fun getAllFilenamesByPath(path: String) : List<String> {
+        val resourcesPath = Paths.get(projectDirAbsolutePath, path)
+        return Files.walk(resourcesPath)
+            .filter { item -> Files.isRegularFile(item) }
+            .map { item -> item.toString().substring(projectDirAbsolutePath.length + path.length + 1) }.toList()
+    }
+
+
+    private fun buildPadding(names: List<String>, padding: Int = 13, separator: String = " | "): String {
+        val result = StringBuilder()
+
+        names.forEach {
+            result.append(it.padEnd(padding, ' '))
+            result.append(separator)
+        }
+        return if (result.isNotEmpty()) result.dropLast(separator.length).toString() else ""
+    }
+
+    private fun processMiniSatSolver(input: String): Boolean {
+        val data = readCnfRequests(input).first()
+
+        with(MiniSatSolver()) {
+            val lit = List(data.vars) { newLiteral() }
+            for (clause in data.clauses)
+                addClause(clause.lit.map { it.sign * lit[abs(it) - 1] })
+
+            return solve()
+        }
+    }
+
+    private fun checkKoSatSolution(ans: List<Int>?, input: String, isSolution: Boolean): Boolean {
+        if (ans == null) return !isSolution // null ~ UNSAT
+
+        val cnfRequest = readCnfRequests(input).first()
+        if (ans.size != cnfRequest.vars) return false
+
+        return cnfRequest.clauses.all { clause -> clause.lit.any { ans.contains(it) } }
+    }
+
+    private fun runTests(path: String) {
+        val filenames = getAllFilenamesByPath(path)
+        println(buildPadding(headerNames))
+        MiniSatSolver()
```

Comment:
Are you creating (and not using) it just to trigger the shared library loading? Then you must deallocate the solver using `.close()` method (so, just call `MiniSatSolver().close();`). Alternative: use `.use{}` construction, since every `Solver` is `AutoCloseable`.
In either way, document this line!

Reply:
OK